File: pixierp/frontend/src/utils/permissions.ts
/**
 * Láthatja-e a felhasználó az árakat (RFQ/rendelés felületek).
 * A backend a profilban can_view_prices flagként küldi (superuser vagy
 * sales.rfqs view/manage jogosultság); a fallback a helyben számolt változat,
 * pl. ha a bejelentkezéskor mentett user objektum még régi.
 *
 * Aki NEM látja az árakat: a /sales/rfqs lista helyett a gyártási oldalakra
 * irányítjuk, és az RFQ adatlap butított módban jelenik meg neki.
 */
export const canViewPrices = (user: any): boolean => {
  if (!user) return false;
  if (user.can_view_prices !== undefined) return !!user.can_view_prices;
  if (user.is_superuser || user.is_staff) return true;
  const perms: any[] = Array.isArray(user.permissions) ? user.permissions : [];
  return perms.some(
    (p: any) =>
      p?.allowed &&
      p.module === 'sales' &&
      (!p.resource || p.resource === 'sales.rfqs') &&
      (p.action === 'view' || p.action === 'manage'),
  );
};

File: pixierp/apps/sales/permissions.py
"""Ár-láthatóság az RFQ/rendelés felületeken.

Aki nem superuser és nincs sales.rfqs view/manage jogosultsága (a gyártói
felhasználók), az a butított RFQ adatlapot kapja: a serializer kihagyja az
ármezőket, és tételt csak a belső leírásig módosíthatja.
"""
from apps.core.permissions import check_permission


def user_can_view_prices(user):
    """Láthatja-e a felhasználó az árakat (superuser vagy sales.rfqs view/manage)."""
    if not user or not getattr(user, 'is_authenticated', False):
        return False
    if user.is_superuser:
        return True
    return check_permission(user, 'sales', 'sales.rfqs', 'view')
